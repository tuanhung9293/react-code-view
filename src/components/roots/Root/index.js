import '../../../../node_modules/video-react/dist/video-react.css'
import React from 'react'
import {BrowserRouter, Route, Switch, Redirect} from 'react-router-dom'
import GeneralErrorBoundary from 'components/errors/GeneralErrorBoundary'
import NotFound from 'components/errors/NotFound'
import ScrollToTop from 'components/ui/ScrollToTop'
import ScreenContent from 'components/ui/ScreenContent'
import Collections from 'components/containers/pages/Collections'
import NasaSearch from 'components/containers/pages/NasaSearch'
import PlayMedia from 'components/containers/pages/PlayMedia'
import EditCollection from 'components/containers/pages/EditCollection'
import AddMedia from 'components/containers/pages/AddMedia'

export function Root() {
  return (
    <GeneralErrorBoundary>
      <BrowserRouter>
        <ScrollToTop>
          <ScreenContent>
            <Switch>
              <Route exact path="/" render={() => (<Redirect to="/collection" />)} />
              <Route path="/collection" component={Collections} />
              <Route path="/nasa-search" component={NasaSearch} />
              <Route path="/play-media/:source/:mediaId" component={PlayMedia} />
              <Route component={NotFound} />
            </Switch>
            <Route path="/collection/edit/:collectionId" component={EditCollection} />
            <Route path="/nasa-search/add-media/:mediaId" component={AddMedia} />
          </ScreenContent>
        </ScrollToTop>
      </BrowserRouter>
    </GeneralErrorBoundary>
  )
}

export default Root