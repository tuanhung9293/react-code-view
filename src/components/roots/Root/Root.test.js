import React from 'react'
import {shallow} from 'enzyme'
import Root from './'


describe('renders', () => {
  it('renders Root', () => {
    expect(shallow(
        <Root />
    )).toMatchSnapshot()
  })
})
