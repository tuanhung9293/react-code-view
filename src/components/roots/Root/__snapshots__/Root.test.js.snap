// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders renders Root 1`] = `
<GeneralErrorBoundary>
  <BrowserRouter>
    <withRouter(ScrollToTop)>
      <ScreenContent>
        <Switch>
          <Route
            exact={true}
            path="/"
            render={[Function]}
          />
          <Route
            component={[Function]}
            path="/collection"
          />
          <Route
            component={[Function]}
            path="/nasa-search"
          />
          <Route
            component={[Function]}
            path="/play-media/:source/:mediaId"
          />
          <Route
            component={[Function]}
          />
        </Switch>
        <Route
          component={[Function]}
          path="/collection/edit/:collectionId"
        />
        <Route
          component={[Function]}
          path="/nasa-search/add-media/:mediaId"
        />
      </ScreenContent>
    </withRouter(ScrollToTop)>
  </BrowserRouter>
</GeneralErrorBoundary>
`;
